### PR TITLE
Load a list of tests to run from a file; connect test output to dif-viewer

### PR DESCRIPTION
**Pull request by rachel-fenichel**
## The basics

- [ ] I branched from develop
- [ ] My pull request is against develop
- [ ] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

^ None of these, but the style guide doesn't currently apply in this directory.

### Proposed Changes

- Use a `test_cases.json` file to specify which tests to run.
- Use a custom reporter for mocha to output test results in a way that's easy to use.
- Use those results in `diff_viewer.html`.
- Make sure folders exist
- Automatically open `diff_viewer.html` after running tests

### CODE REVIEW

File: tests/screenshot/diff-reporter.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Reporter that prints results to the console with the same
+ * format as the spec reporter, but also saves a test_output.js file with a
```
Comment (RoboErikG) [inline tests/screenshot/diff-reporter.js]: "with a / that just wraps" you accidentally a noun.

File: tests/screenshot/diff_screenshots.js [hunk 2]
```diff
@@ -31,35 +31,55 @@ var old_dir = 'tests/screenshot/outputs/old/';
...
-// - iterate over all test cases
-// - use the name of the test case file as the name of the test
-// - improve doneReading to not rely on globals like this
-// - write or find basic html page that lets you open pairs of outputs + diffs
+if (!fs.existsSync(diff_dir)){
+  fs.mkdirSync(diff_dir);
+}
+
+function getTestList() {
+  var file = fs.readFileSync('tests/screenshot/test_cases/test_cases.json');
```
Comment (RoboErikG) [inline tests/screenshot/diff_screenshots.js]: Should this path be up with the new dir?

File: tests/screenshot/gen_screenshots.js [hunk 3]
```diff
@@ -34,25 +40,39 @@ module.exports = genScreenshots;
...
+  for (var i = 0, testName; testName = test_list[i]; i++) {
+    await genSingleScreenshot(browser_new, 'new', testName);
+    await genSingleScreenshot(browser_old, 'old', testName);
+  }
 
   await cleanUp(browser_new, browser_old);
   return 0;
 }
 
+function getTestList() {
```
Comment (RoboErikG) [inline tests/screenshot/gen_screenshots.js]: This looks like the same code as in diff_screenshots.js. Should some of this get pulled out into a utils file?

File: tests/screenshot/diff_screenshots.js (hunk 2 above)
Comment (rachel-fenichel) [reply tests/screenshot/diff_screenshots.js]: Done.

File: tests/screenshot/diff-reporter.js (hunk 1 above)
Comment (rachel-fenichel) [reply tests/screenshot/diff-reporter.js]: Done.

File: tests/screenshot/gen_screenshots.js (hunk 3 above)
Comment (rachel-fenichel) [reply tests/screenshot/gen_screenshots.js]: Maybe?  It's only this function.  Definitely if I end up adding more shared code.

**Comment by rachel-fenichel** [comment]
Added a few things I remembered during lunch:
- gitignore
- Check if the old screenshot exists, and skip if so

**Pull request merged**